source/extensions/filters/listener/proxy_protocol/proxy_protocol.h

```diff
@@ -66,24 +101,26 @@ class Config : public Logger::Loggable<Logger::Id::filter> {
   bool isPassThroughTlvTypeNeeded(uint8_t type) const;
 
   /**
-   * Filter configuration that determines if we should pass-through requests without
-   * proxy protocol. Should only be configured to true for trusted downstreams.
+   * Return true if the given PROXY protocol version should be parsed by the filter.
    */
-  bool allowRequestsWithoutProxyProtocol() const;
+  bool isVersionAllowed(ProxyProtocolVersion version) const;
```

Comment:
This is always called with a compile-time-specified version; just make two functions `isVersion1Allowed` and `isVersion2Allowed`.

Reply:
Good point, done